nomos-mix/core/src/message_blend/mod.rs

```diff
@@ -10,41 +10,64 @@ use std::pin::Pin;
 use std::task::{Context, Poll};
 pub use temporal::TemporalProcessorSettings;
 
+use crate::membership::Membership;
 use crate::message_blend::crypto::CryptographicProcessor;
 use crate::message_blend::temporal::TemporalProcessorExt;
 use crate::MixOutgoingMessage;
+use nomos_mix_message::MixMessage;
+use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
 use tokio::sync::mpsc;
 use tokio::sync::mpsc::UnboundedSender;
 use tokio_stream::wrappers::UnboundedReceiverStream;
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct MessageBlendSettings {
-    pub cryptographic_processor: CryptographicProcessorSettings,
+pub struct MessageBlendSettings<M>
+where
+    M: MixMessage,
+    M::PrivateKey: Serialize + DeserializeOwned,
+{
+    pub cryptographic_processor: CryptographicProcessorSettings<M::PrivateKey>,
     pub temporal_processor: TemporalProcessorSettings,
 }
 
 /// [`MessageBlendStream`] handles the entire mixing tiers process
 /// - Unwraps incoming messages received from network using [`CryptographicProcessor`]
 /// - Pushes unwrapped messages to [`TemporalProcessor`]
-pub struct MessageBlendStream<S, Rng> {
+pub struct MessageBlendStream<S, Rng, M>
+where
+    M: MixMessage,
+{
     input_stream: S,
     output_stream: BoxStream<'static, MixOutgoingMessage>,
     temporal_sender: UnboundedSender<MixOutgoingMessage>,
-    cryptographic_processor: CryptographicProcessor,
+    cryptographic_processor: CryptographicProcessor<Rng, M>,
     _rng: PhantomData<Rng>,
 }
 
-impl<S, Rng> MessageBlendStream<S, Rng>
+impl<S, Rng, M> MessageBlendStream<S, Rng, M>
 where
     S: Stream<Item = Vec<u8>>,
     Rng: RngCore + Unpin + Send + 'static,
+    M: MixMessage,
+    M::PrivateKey: Serialize + DeserializeOwned,
+    M::PublicKey: Clone + PartialEq,
 {
-    pub fn new(input_stream: S, settings: MessageBlendSettings, rng: Rng) -> Self {
-        let cryptographic_processor = CryptographicProcessor::new(settings.cryptographic_processor);
+    pub fn new(
+        input_stream: S,
+        settings: MessageBlendSettings<M>,
+        membership: Membership<M>,
+        cryptographic_processor_rng: Rng,
+        temporal_processor_rng: Rng,
```

Comment:
We need two Rngs. One for crypto processor, and the other for the temporal processor.

Reply:
1. Aditionally we could spawn 2 rngs from a single rng source passed here. But I do not think it is important atm.

Reply:
removed the leftover